api/swagger.yaml

```diff
@@ -0,0 +1,788 @@
+openapi: 3.0.1
+info:
+  title: Space Tycoon
+  description: Space Tycoon server.
+  contact:
+    url: https://www.garage-trip.cz/
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+externalDocs:
+  description: Find out more about Space Tycoon game
+  url: https://github.com/gdg-garage/space-tycoon
+servers:
+- url: https://space-tycoon.garage-trip.cz/
+tags:
+- name: login
+  description: Get user session
+- name: staticdata
+  description: Static game state (planets)
+- name: data
+  description: "Dynamic game data (scores, prices, spaceship positions)"
+- name: commands
+  description: Specify commands for your ships
+- name: tick
+  description: Signal that your turn is over for the current tick
+paths:
+  /login:
+    post:
+      tags:
+      - login
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/Credentials'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content: {}
+        "403":
+          description: Invalid name or password
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "405":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /staticdata:
+    post:
+      tags:
+      - staticdata
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StaticData'
+  /data:
+    post:
+      tags:
+      - data
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Data'
+  /commands:
+    post:
+      tags:
+      - commands
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/Commands'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content: {}
+        "405":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /tick:
+    post:
+      tags:
+      - tick
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/Tick'
+        required: true
+      responses:
+        "200":
+          description: OK
+          content: {}
+        "405":
+          description: Invalid input
+          content: {}
+components:
+  schemas:
+    Command:
+      required:
+      - type
+      type: object
+      properties:
+        type:
+          type: string
+      discriminator:
+        propertyName: Type_
+      oneOf:
```

Comment:
TL;DR: Imho we should. No having repairs could make aggressive strategies too easy and too powerful. And if there are too many ships dying then trading becomes very hard. 

Imho it makes sense to either:
- allow any ship to repair itself anywhere (this way the repairs would have to be kinda slow so that it's not very easy to repair all of your ships)
- allow shipyard ships to repair other ships (this way the repairs could be super fast because they are constrained by number of shipyards player has)

Extra notes:

**Dangers of no repairs**
Imho could be hard to defend yourself effectively. E.g. someone can send ships to attack your mothership or cargo ships and gradually kill them because of lack of repair. It could be really hard to write a meaningful defense against these "wear down" strategies. If there are too many ships dying then trading becomes very hard.

**Issues with automatic passive regeneration**
I was considering if automatic regeneration amount would work. It could be hard for us to tune the regeneration amount imho. Also making ships "wait" so that they repair in safety without setting any command on them is weird - it requires bots to remember that they have the ship waiting on purpose.

Reply:
Should repairs be paid?

Reply:
moving into separate issue: https://github.com/gdg-garage/space-tycoon/issues/25